Allow microvm configuration from a single json passed as command line parameter (#1212)
firecracker-microvm/firecracker: docs/getting-started.md

    }'
```

#### Configuring the microVM without sending API requests

If you'd like to boot up a guest machine without using the API socket, you can do that 

> by passing the parameter `config-file` to the Firecracker process. The command for 

Reviewer: ```suggestion
by passing the parameter `--config-file` to the Firecracker process. The command for 
```

starting Firecracker with this option will look like this:

```bash
./firecracker --api-sock /tmp/firecracker.socket --config-file 
<path_to_the_configuration_file>
```    

`path_to_the_configuration_file` should represent the path to a file that contains a 
JSON which stores the entire configuration for all of the microVM's resources. You 

> **must** provide in your JSON the configuration for the guest kernel and rootfs, as 

Reviewer: nit: I'd rephrase to:
> The JSON **must** contain the configuration ... blah

these are mandatory, but all of the other resources are optional, so it's your choice 
if you want to configure them or not. Because using this configuration method will 
also start the microVM, you need to specify all desired pre-boot configurable resources 
in that JSON. The names of the resources are the ones from the `firecracker.yaml` file 
and the names of their fields are the same that are used in API requests. 

> You can find an example of configuration file, named `vm_config.json`, in `tests/framework`.

Reviewer: ```suggestion
You can find an example of configuration file, at `tests/framework/vm_config.json`.
```

    }'
```

#### Configuring the microVM without sending API requests

If you'd like to boot up a guest machine without using the API socket, you can do that 
by passing the parameter `vmm-config` to the Firecracker process. The command for starting 
Firecracker with this option will look like this:

```bash
./firecracker --api-sock /tmp/firecracker.socket --vmm-config "JSON_BODY"
```    

"JSON_BODY" should contain the entire configuration for all of the microVM's resources. 
You **must** provide in your JSON the configuration for the guest kernel and rootfs, as 

> these are mandatory, but all of the other resources are optional, so it's your choice if 

Reviewer: Clearly specify that using this configuration method will also start the microVM with said configuration, so that all desired pre-boot configurable resources need to be specified.
